### ISSUE TITLE
Eliciting Latent Predictions from Transformers with the Tuned Lens

### ISSUE THREAD

**Issue by tatsuropfgt** [InabaTatsuro/papers#75]
# Eliciting Latent Predictions from Transformers with the Tuned Lens [Belrose+'23]

<img width="557" alt="Screenshot 2024-02-09 at 0 05 29" src="https://github.com/InabaTatsuro/papers/assets/102784221/1ed80713-d93d-4342-b6e7-932697a20e3f">

## Abst
- proposed tuned lens, the improved version of LogitLens which is used for transforming hidden states into logits

## Logit Lens

<img width="212" alt="Screenshot 2024-02-09 at 0 02 58" src="https://github.com/InabaTatsuro/papers/assets/102784221/72e123a2-e5af-4ffd-ae63-584799fcd351">

- l layer -> l+1 layer

<img width="466" alt="Screenshot 2024-02-09 at 0 02 49" src="https://github.com/InabaTatsuro/papers/assets/102784221/be4546c5-e7f7-43f8-8e7c-6248bf35e8e2">

- l layer -> logits

<img width="349" alt="Screenshot 2024-02-09 at 0 03 50" src="https://github.com/InabaTatsuro/papers/assets/102784221/12ab6044-5512-4382-be13-861a6f657d98">

- LogitLens: residuals to zero
- well for GPT2, bad for GPT-Neo family

<img width="463" alt="Screenshot 2024-02-09 at 0 05 01" src="https://github.com/InabaTatsuro/papers/assets/102784221/88d5de12-7262-4a87-805f-8c8d544bb1c1">

- LogitLens extension: add the last transformer layer
- partially successful

### Flaws
- unreliability
  - LogitLens can't elicit predictions from other models
  - for BLOOM and OPT 125M, the top 1 prediction of the logit lens is often the input token

<img width="338" alt="Screenshot 2024-02-09 at 0 24 16" src="https://github.com/InabaTatsuro/papers/assets/102784221/6ea3896d-c06b-4027-9e64-1d87b0b94330">

- If q_l(LogitLens) is an "unbiased estimator", the probability of each token is same between q and p(model prediction)
  - But **biased**
- measure the degree of bias using the KL divergence

## Tuned Lens
<img width="408" alt="Screenshot 2024-02-09 at 0 36 37" src="https://github.com/InabaTatsuro/papers/assets/102784221/05c9c630-fd70-4412-9be2-35129c39703a">

- add learnable constant value b_l

<img width="473" alt="Screenshot 2024-02-09 at 0 49 50" src="https://github.com/InabaTatsuro/papers/assets/102784221/4cf27bc1-fad6-4c51-93e4-71ebc420d3d7">

- rogue dimensions lead to be distributed unevenly across layers
- Layer4 introduces two outlier(top), removing them(bottom)

<img width="483" alt="Screenshot 2024-02-09 at 0 53 23" src="https://github.com/InabaTatsuro/papers/assets/102784221/9a75239f-6159-435f-89c2-ee36381ac4cf">

- TunedLens
  - A_l for mapping from layer l to final layer
  - b_l for removing bias of distribution

## Results
<img width="436" alt="Screenshot 2024-02-09 at 0 29 34" src="https://github.com/InabaTatsuro/papers/assets/102784221/5b1e49d8-37cd-4ce2-b5ad-a00d4beca50e">

- Bias (difference of distributions)

<img width="723" alt="Screenshot 2024-02-09 at 0 29 19" src="https://github.com/InabaTatsuro/papers/assets/102784221/d6d4a870-0a27-4946-ae31-d5af4c61bfd3">
